tools/inc/stljobs.h

```diff
@@ -0,0 +1,606 @@
+    (void) hJob.detach();
+}
+
+class no_input_pipe {
+public:
+    no_input_pipe() {
+        SECURITY_ATTRIBUTES inheritSa;
+        inheritSa.nLength              = sizeof(inheritSa);
+        inheritSa.lpSecurityDescriptor = nullptr;
+        inheritSa.bInheritHandle       = TRUE;
+        HANDLE read;
+        if (!CreatePipe(&read, hDevNull.out(), &inheritSa, 0)) {
+            api_failure("CreatePipe");
+        }
+
+        CloseHandle(read);
+    }
+
+    no_input_pipe(const no_input_pipe&) = delete;
+    no_input_pipe& operator=(const no_input_pipe&) = delete;
+
+    HANDLE get() const noexcept {
+        return hDevNull.get();
+    }
+
+    static const no_input_pipe& instance() {
+        static no_input_pipe instance_;
+        return instance_;
+    }
+
+private:
+    handle<null_handle_policy> hDevNull;
+};
+
+class tp_io {
+public:
+    tp_io() = default;
+
+    explicit tp_io(handle<invalid_handle_value_policy>&& fileHandle_, PTP_WIN32_IO_CALLBACK callback, void* const pv,
+        const PTP_CALLBACK_ENVIRON pcbe)
+        : io(CreateThreadpoolIo(fileHandle_.get(), callback, pv, pcbe)) {
+        if (!io) {
+            api_failure("CreateThreadpoolIo");
+        }
+
+        fileHandle = std::move(fileHandle_);
+    }
+
+    tp_io(tp_io&& other) noexcept : fileHandle(std::move(other.fileHandle)), io(std::exchange(other.io, nullptr)) {}
+
+    tp_io& operator=(tp_io&& other) noexcept {
+        tp_io moved{std::move(other)};
+        swap(moved, *this);
+        return *this;
+    }
+
+    friend void swap(tp_io& lhs, tp_io& rhs) noexcept {
+        using std::swap;
+        swap(lhs.fileHandle, rhs.fileHandle);
+        swap(lhs.io, rhs.io);
+    }
+
+    [[nodiscard]] HANDLE get_file() const noexcept {
+        return fileHandle.get();
+    }
+
+    void start_threadpool_io() noexcept {
+        StartThreadpoolIo(io);
+    }
+
+    void cancel_threadpool_io() noexcept {
+        CancelThreadpoolIo(io);
+    }
+
+    explicit operator bool() const noexcept {
+        return io != nullptr;
+    }
+
+    handle<invalid_handle_value_policy> close() noexcept {
+        assert(io != nullptr);
+        WaitForThreadpoolIoCallbacks(io, TRUE);
+        CloseThreadpoolIo(io);
+        return std::move(fileHandle);
+    }
+
+    void wait(bool cancelPending) noexcept {
+        WaitForThreadpoolIoCallbacks(io, cancelPending);
+    }
+
+private:
+    handle<invalid_handle_value_policy> fileHandle{};
+    PTP_IO io{};
+};
+
+struct output_collecting_pipe {
+    static constexpr unsigned long kernelBufferSize = 4096;
+    static constexpr unsigned long bufferSize       = 4096;
+
+    output_collecting_pipe() {
+        // generate a random name for the pipe (we must use a named pipe because anonymous pipes from CreatePipe can't
+        // be used in asynchronous mode)
+        constexpr size_t pipeNameBufferCount        = 15 + 8 * 8 + 1;
+        wchar_t pipeNameBuffer[pipeNameBufferCount] = L"\\\\.\\pipe\\Local\\";

```

Comment:
And possibly a guide-comment, so that we don't have to count :-)
````c++
//                                                123456789012345                                        
wchar_t pipeNameBuffer[pipeNameBufferCount] = LR"(\\.\pipe\Local\)";
````